js/utils/getBitcoinPrice.js

```diff
@@ -121,21 +135,42 @@ module.exports = function(currency, callback){
           });
     };
 
+    if(window.btcAverages.timeStamp && Math.floor((new Date() - window.btcAverages.timeStamp)/60000) < 15){
+      typeof callback === 'function' && callback(window.btcAverages.rates[currency]);
+    } else {
+      callBlockchain();
+    }
 
     var makeAveragePrice = function ()
     {
-      var sum = 0,
-          btAve = 0;
-      for (var i = 0; i < btPrices.length; i++)
-      {
-        sum = sum + Number(btPrices[i]);
+      btcAverages.timeStamp = new Date();
+      var keys = {};
+      for (var i in btPrices) {
+        keys = $.extend(keys, btPrices[i]);
```

Comment:
You still need the hasOwnProperty check here, and below, in all for in loops. Sorry for not flagging these the last time.